src/olmo_core/data/numpy_dataset.py

```diff
@@ -1519,6 +1652,8 @@ def build(self) -> NumpyDatasetBase:
                 paths.extend(matches)
         elif self.paths:
             paths = self.paths
+        elif self.source_mixture_config:
+            log.info("Building dataset from source mixture...")
```

Comment:
`self.source_mixture_config` is only used when `self.type == NumpyDatasetType.fsl`, right? We should check for that, otherwise looks like we could get a silent failure where `paths` just ends up empty.

Reply:
Yep, good catch. On it.